src/parser.ml

```diff
@@ -1673,12 +1788,17 @@ let link_reference_definition st =
   if next st <> ':' then raise Fail;
   ws st;
   let destination = link_destination st in
+  let pos = pos st in
+  ws st;
+  let attributes = inline_attribute_string st in
+  if attributes = empty_attributes then
+    set_pos st pos;
```

Comment:
This logic of resetting the position should go into `inline_attribute_string`, so that if there are no attributes the position is not changed.